tests/pipeline/test_engine.py

```diff
@@ -1529,3 +1529,30 @@ def test_run_chunked_pipeline(self):
             chunksize=22
         )
         self.assertTrue(chunked_result.equals(pipeline_result))
+
+
+class MaximumRegressionTest(WithSeededRandomPipelineEngine,
+                            ZiplineTestCase):
+    ASSET_FINDER_EQUITY_SIDS = (1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
+
+    def test_no_groupby_maximum(self):
+        # This is a regression test for a bug where factor.top(1) would fail
+        # when not passed a groupby parameter.
+
+        factor = TestingDataSet.float_col.latest
+        maximum = factor.top(1)
+        pipe = Pipeline({'factor': factor, 'maximum': maximum})
```

Comment:
Any reason for including the actual factor in the pipeline?

Reply:
The line below that does `result.groupby(level=0).max()` is using the factor to re-compute the max to verify the result that came out of pipeline.